blocks/color-palette/index.js

```diff
@@ -1,26 +1,94 @@
 /**
  * External dependencies
  */
-import { CirclePicker } from 'react-color';
+import classnames from 'classnames';
+import { ChromePicker } from 'react-color';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { Popover } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
+import './style.scss';
 import withEditorSettings from '../with-editor-settings';
 
-function ColorPalette( { colors, onChange, value, withTransparentOption } ) {
-	if ( withTransparentOption ) {
-		colors = [ ...colors, 'transparent' ];
+class ColorPalette extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			opened: false,
+		};
+		this.openPicker = this.openPicker.bind( this );
+		this.closePicker = this.closePicker.bind( this );
+	}
+
+	openPicker() {
+		this.setState( { opened: true } );
 	}
 
-	return (
-		<CirclePicker
-			color={ value }
-			colors={ colors }
-			onChangeComplete={ ( colorValue ) => onChange( colorValue ) }
-			style={ { marginBottom: '20px' } }
-		/>
-	);
+	closePicker() {
+		this.setState( { opened: false } );
+	}
+
+	render() {
+		const { colors, value, onChange } = this.props;
+		return (
+			<div className="blocks-color-palette">
+				{ colors.map( ( color ) => {
+					const style = { color: color };
+					const className = classnames( 'blocks-color-palette__item', { 'is-active': value === color } );
+					/* Disable reason: aria-current seems like a valid aria-prop and well-suited for this */
+					/* eslint-disable jsx-a11y/aria-props */
+					return (
+						<div key={ color } className="blocks-color-palette__item-wrapper">
+							<button
+								className={ className }
+								style={ style }
+								onClick={ () => onChange( value === color ? undefined : color ) }
+								aria-label={ __( 'Color: ' ) + color }
```

Comment:
We should use variable substitution here:

```js
sprintf( __( 'Color: %s', color )
```

>- Do not leave leading or trailing whitespace in a translatable phrase.
>- Add the variables as placeholders to the string as in some languages the placeholders change position.

https://developer.wordpress.org/themes/functionality/internationalization/#best-practices-for-writing-strings